src/twister2/log_parser/console_log_parser.py

```diff
@@ -1,20 +1,106 @@
+"""
+Console log parser
+"""
 from __future__ import annotations
 
-from dataclasses import dataclass
-from typing import Iterator
+import logging
+import re
+import time
+from typing import Callable, Iterator
 
+from twister2.exceptions import TwisterHarnessParserException
+from twister2.log_parser import SubTestResult
 from twister2.log_parser.log_parser_abstract import LogParserAbstract
 
+logger = logging.getLogger(__name__)
 
-@dataclass
-class HarnessConfig:
-    fail_on_fault: bool = False
+ZEPHYR_FATAL_ERROR: str = 'ZEPHYR FATAL ERROR'
+
+ONE_LINE: str = 'one_line'
+MULTI_LINE: str = 'multi_line'
 
 
-# TODO: to be implemented
 class ConsoleLogParser(LogParserAbstract):
     """Console log parser."""
 
     def __init__(self, stream: Iterator[str], *, harness_config: dict, **kwargs):
         super().__init__(stream)
         self.harness_config = harness_config
+        self.matched_lines: list[str] = []
+        self.type: str = harness_config.get('type')
+        self.ordered: bool = harness_config.get('ordered', False)
+        self.regex: list[str] = harness_config.get('regex', [])
+        self.repeat: int = harness_config.get('repeat', 1)
```

Comment:
Yes, it wasn't implemented in V1. I am wondering if we need it. I can implemented it or add TODO comment that we want to have such feature in the future.

Reply:
I am opting for dropping this line. It is not used anywhere and it is also not clear what is a reasoning behind it. This will just artificially decrease the code coverage.  We can always add it when a clear feature is requested.

Reply:
Removed